### DISCUSSION TITLE
city isn't included in search & lookup endpoint, but in details endpoint

### DISCUSSION THREAD

**Discussion by jtopenpetition** [osm-search/Nominatim#3187] [Q&A] [answered]
We have this very specific case of missing "city". The result contains the city only in the display_name field, but not in the address field:

https://nominatim.openstreetmap.org/search.php?street=Postwai+1&city=Nebel&country=Germany&postalcode=25946&format=jsonv2&addressdetails=1
https://nominatim.openstreetmap.org/lookup?osm_ids=N668984942&format=json&addressdetails=1

However, the city is there, if we use the details api ("for debugging"):

https://nominatim.openstreetmap.org/details.php?osmtype=N&osmid=668984942&addressdetails=1&format=json

Any idea why the city is excluded from the address field in the "non-debugging" responses?

**Comment by mtmail** [comment, marked as answer]
Nominatim assigns https://www.openstreetmap.org/relation/1428589 (admin_level=7) to the field "municipality" but then the field 'city' and 'municipality' usually mean the same and only one gets returned. With many places in the world it's not clear which of the two fields is more relevant to the user.

The 'geocodejson' output has improved logic https://nominatim.openstreetmap.org/lookup?osm_ids=N668984942&format=geocodejson&addressdetails=1

**Reply by jtopenpetition** [reply]
thank you for your answer. this really helps. I was somewhat aware, that different output formats contain different data, but didn't imagine, that the quality is also different ;o/ So again, thank you ;o)
